e2e/tests-dfx/ledger.bash

```diff
@@ -210,6 +210,13 @@ tc_to_num() {
   assert_match "Refunded at block height"
   assert_match "with message: Subnet $SUBNET_ID does not exist"
 
+  # Verify that registry is queried before sending any ICP to CMC
+  CANISTER_ID="aaaaa-aa"
+  balance=$(dfx ledger balance)
+  assert_command_fail dfx ledger create-canister --amount=100 --next-to "$CANISTER_ID" "$(dfx identity get-principal)"
+  assert_contains "[Registry] routing table not found" 
```

Comment:
It's not set up properly. As a matter of fact, me and @sesi200 did not find a better way to test this with the NNS extension. However, it still shows that it does not transfer any funds out of the users account; with the old implementation, this test would fail on the next line.

Reply:
> did not find a better way to test this with the NNS extension

This means that you cannot have a successful test for `--next-to`, right?

Reply:
Given a `subnet_id` of the local replica, you can pass this
```
Encode!(RegistryCanisterInitPayloadBuilder::new().with_initial_mutations(invariant_compliant_mutation_with_subnet_id(0, subnet_id, None)).build()).unwrap()
```
with `RegistryCanisterInitPayloadBuilder` and `invariant_compliant_mutation_with_subnet_id` from the dfinity/ic monorepo as the initial argument of the registry canister.

Reply:
Yes, in fact, there exist no tests, except this one, for `--next-to`. This has only ever been tested on mainnet. Thanks for the instructions, but I'm not sure I understand them, since I'm using the dfx NNS extension and no rust code; I assume you're proposing a fix on a deeper level. If you have time, we can make a call next week to implement this wherever it belongs :)

Reply:
> on a deeper level

this would be a fix of the dfx nns extension itself (which is written in Rust and would need to depend on the dfinity/ic monorepo unless it already does)

Reply:
> a fix of the dfx nns extension itself

or even better of ic-nns-init - I have an idea - let me give it a quick try

Reply:
Adding
```
--registry-local-store-dir ~/.local/share/dfx/network/local/state/replicated_state/ic_registry_local_store
```
to ic-nns-init [here](https://github.com/dfinity/dfx-extensions/blob/a37848b4e3ccf563ac2b77fbf6193a4174dfa0e3/extensions/nns/src/install_nns.rs#L419) should fix the dfx nns extension.

Reply:
The only challenge I see is if the path `~/.local/share/dfx/network/local/state/replicated_state/ic_registry_local_store` works across platforms.

Reply:
Dropped the assert on registry being misconfigured